tests/fast-integration/monitoring/helpers.js

```diff
@@ -110,9 +122,135 @@ async function assertTimeSeriesExist(metric, labels) {
   assert.isEmpty(resultlessQueries, `Following queries return no results: ${resultlessQueries.join(", ")}`)
 }
 
+async function retryUrl(url, redirectURL, httpStatus) {
+  let retries = 0
+  while (retries < 20) {
+    let res = await queryGrafana(url, redirectURL, httpStatus)
+    if (res === true) {
+      return res
+    }
+    await sleep(5*1000)
+    retries++
+  }
+  return false
+}
+
+async function assertGrafanaredirect(redirectURL) {
+  let vs = await getVirtualService("kyma-system", "monitoring-grafana")
+  let url = "https://"+vs
+  if (redirectURL.includes("https://dex.")) {
+    console.log("Checking redirect for dex")
+    return await retryUrl(url, redirectURL, 200)
+  }
+
+  if (redirectURL.includes("https://kyma-project.io/docs")) {
+    console.log("Checking redirect for kyma docs")
+    return await retryUrl(url, redirectURL, 403)
+  }
+
+  if (redirectURL.includes("https://accounts.google.com/signin/oauth")) {
+    console.log("Checking redirect for google")
+    return await retryUrl(url, redirectURL, 200)
+  }
+
+  if (redirectURL.includes("grafana")) {
+    console.log("Checking redirect for grafana")
+    return await retryUrl(url, redirectURL, 200)
+  }
+}
+
+async function createSecret() {
+  const sec = {
+    apiVersion: "v1",
+    kind: "Secret",
+    metadata: {
+      name: "monitoring-auth-proxy-grafana-user",
+      namespace: "kyma-system",
+    },
+    type: "Opaque",
+    data: {
+      OAUTH2_PROXY_SKIP_PROVIDER_BUTTON: toBase64("true")
+    },
+  }
+  console.log("Creating secret: monitoring-auth-proxy-grafana-user ")
+  await k8sApply([sec], "kyma-system");
+}
+
+async function updateProxyDeployment() {
```

Comment:
I am wondering if we should configure at least the trusted-ip range in the secret to get as close as possible to the way a user would configure this.

Reply:
I tried it with passing the trusted IP in secret it did not work out